src/app/mandate/mandate.component.ts

```diff
@@ -1,30 +1,67 @@
-import {Component} from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import {ComponentsModule} from "@biggive/components-angular";
 import {DatePipe} from "@angular/common";
 import {Mandate} from "../mandate.model";
 import {ActivatedRoute} from "@angular/router";
 import {MoneyPipe} from "../money.pipe";
 import {myRegularGivingPath} from '../app-routing';
-
+import {MatProgressSpinner} from '@angular/material/progress-spinner';
+import { RegularGivingService } from '../regularGiving.service';
 
 @Component({
     selector: 'app-mandate',
-    imports: [
-        ComponentsModule,
-        DatePipe,
-        MoneyPipe
-    ],
+  imports: [
+    ComponentsModule,
+    DatePipe,
+    MoneyPipe,
+    MatProgressSpinner
+  ],
     templateUrl: './mandate.component.html',
     styleUrl: './mandate.component.scss'
 })
-export class MandateComponent {
+
+export class MandateComponent implements OnInit {
   protected mandate!: Mandate;
   protected readonly cancelPath;
+  private mandateRefreshTimer: number|undefined;
+  private timePageLoaded = new Date();
 
   constructor(
-    private route: ActivatedRoute
+    private route: ActivatedRoute,
+    private regularGivingService: RegularGivingService,
   ) {
     this.mandate = this.route.snapshot.data.mandate;
     this.cancelPath = `/${myRegularGivingPath}/${this.mandate.id}/cancel`;
   }
+
+  ngOnInit(): void {
+    if (this.mandate.status === 'pending') {
+      this.pollForMandateUpdate();
+    }
+  }
+
+  private pollForMandateUpdate() {
+    this.mandateRefreshTimer = window.setInterval(
+      async () => {
+        const updatedMandate = await this.regularGivingService.getActiveMandate(this.mandate.id).toPromise();
+
+        if (updatedMandate && updatedMandate.status !== 'pending') {
+          this.mandate = updatedMandate;
+          window.clearInterval(this.mandateRefreshTimer);
+        }
+
+        if (this.timedOut) {
+          window.clearInterval(this.mandateRefreshTimer);
+        }
+      },
+      5_000
```

Comment:
```suggestion
      2_000
```
Maybe? 5s minimum seems a little long to have to wait.